internal/painter/gl/draw.go

```diff
@@ -26,6 +27,79 @@ func (p *painter) defineVertexArray(prog Program, name string, size, stride, off
 	p.logError()
 }
 
+func (p *painter) drawBlur(b *canvas.Blur, pos fyne.Position, frame fyne.Size) {
+	if b.Radius == 0 {
+		return
+	}
+	radius := b.Radius * p.pixScale
+
+	w := roundToPixel(frame.Width*p.pixScale, 1.0)
+	h := roundToPixel(frame.Height*p.pixScale, 1.0)
+	pixels := make([]uint8, int(w*h)*4)
+
+	p.ctx.ReadPixels(0, 0, int(w), int(h), colorFormatRGBA, unsignedByte, pixels)
+	p.logError()
+
+	img := &captureImage{
+		pix:    pixels,
+		width:  int(w),
+		height: int(h),
+	}
+
+	x := roundToPixel(pos.X*p.pixScale, 1.0)
+	y := roundToPixel(pos.Y*p.pixScale, 1.0)
+	w = roundToPixel(b.Size().Width*p.pixScale, 1.0)
+	h = roundToPixel(b.Size().Height*p.pixScale, 1.0)
+	bounds := image.Rect(int(x), int(y), int(x+w), int(y+h))
+	crop := img.SubImage(bounds)
+	texture := p.imgToTexture(crop, canvas.ImageScaleFastest)
+
+	p.ctx.ActiveTexture(texture0)
+	p.ctx.BindTexture(texture2D, texture)
+	p.logError()
+
+	points := p.rectCoords(b.Size(), pos, frame, canvas.ImageFillStretch, 1.0, 0)
+	p.ctx.UseProgram(p.blurProgram)
+	vbo := p.createBuffer(points)
+	p.defineVertexArray(p.blurProgram, "vert", 3, 5, 0)
+	p.defineVertexArray(p.blurProgram, "vertTexCoord", 2, 5, 3)
+
+	p.ctx.BlendFunc(srcAlpha, oneMinusSrcAlpha)
+	p.logError()
+
+	vertAttrib := p.ctx.GetUniformLocation(p.blurProgram, "radius")
+	p.ctx.Uniform1f(vertAttrib, radius)
+
+	vertAttrib = p.ctx.GetUniformLocation(p.blurProgram, "size")
+	p.ctx.Uniform2f(vertAttrib, w, h)
+
+	sum := float32(0.0)
+	length := int(radius)*2 + 1
+	values := make([]float32, length)
```

Comment:
Same here re. allocation at 60 fps. But at least this one is small, compared to the pixbuf. But we could (and maybe should) still take an approach like storing this slice on the canvas.Blur itself so the allocated space can be reused next time.